src/widget/wbeatspinbox.cpp

```diff
@@ -236,3 +242,22 @@ QValidator::State WBeatSpinBox::validate(QString& input, int& pos) const {
 
     return QValidator::Acceptable;
 }
+
+bool WBeatSpinBox::event(QEvent* pEvent) {
+    if (pEvent->type() == QEvent::ToolTip) {
+        updateTooltip();
+    } else if (pEvent->type() == QEvent::FontChange) {
+        const QFont& fonti = font();
+        qDebug() << "WBeatSpinBox::event QEvent::FontChange" << fonti.pixelSize() * m_scaleFactor;
+        // Change the new font on the fly by casting away its constancy
+        // using setFont() here, would results into a recursive loop
+        // resetting the font to the original css values.
+        // Only scale pixel size fonts, point size fonts are scaled by the OS
+        if (fonti.pixelSize() > 0) {
+            const_cast<QFont&>(fonti).setPixelSize(fonti.pixelSize() * m_scaleFactor);
+        }
+        // TODO(Be): Figure out how to apply newly scaled font???
```

Comment:
The repopulation in WEffectSelector was there before the resizing hack. 
It is required to elide the text according to the new lenght measure.